students/perrymant/22-Logging.md

```diff
@@ -0,0 +1,124 @@
+    - `ERROR`: Error log messages are used to relay errors that occur at various levels in a computer system. Unfortunately, many error messages only give you a starting point as to why they occurred.
+    - `ALERT`: An alert is meant to indicate that something interesting has happened.
+
+- The levels for `java.util.logging` (the underlying loggin framework in Java) in descending order are:
+    - `SEVERE` (highest value): indicating a serious failure
+    - `WARNING`: indicating a potential problem
+    - `INFO`: a message level for informational messages.
+    - `CONFIG`: a message level for static configuration messages, to assist in debugging problems that may be associated with particular configurations.
+    - `FINE`, `FINER`, `FINEST`  (lowest value)
+    All of `FINE`, `FINER`, and `FINEST` are intended for relatively detailed tracing.  The exact meaning of the three levels will vary between subsystems, but in general, `FINEST` should be used for the most voluminous detailed output, `FINER` for somewhat less detailed output, and `FINE` for the  lowest volume (and most important) messages.
+
+- In Syslog, the logging framework for OS and Networking, the levels are:
+
+| Severity      | Description                       | Condition                                                                              |
+|---------------|-----------------------------------|----------------------------------------------------------------------------------------|
+| Emergency     | System is unusable                | A panic condition.                                                                     |
+| Alert         | Action must be taken immediately  | A condition that should be corrected immediately, such as a corrupted system database. |
+| Critical      | Critical conditions               | Hard device errors.                                                                    |
+| Error         | Error conditions                  |                                                                                        |
+| Warning       | Warning conditions                |                                                                                        |
+| Notice        | Normal but significant conditions | Conditions that are not error conditions, but that may require special handling.       |
+| Informational | Informational messages            |                                                                                        |
+| Debug         | Debug-level messages              | Messages that contain information normally of use only when debugging a program.       |
+(source [Wiki](https://en.wikipedia.org/wiki/Syslog))
```

Comment:
I think you need a space here because is showed on a new row. Or you could put above so:

- In Syslog, the logging framework for OS and Networking, the log levels are (source [wiki](http://...)):